examples/MLIRGPU/gpu-matmul.mlir

```diff
@@ -0,0 +1,84 @@
+module attributes {gpu.container_module} {
+  gpu.module @kernels {
+    gpu.func @mutmal(%matA: memref<1024x1024xf32>, %matB: memref<1024x1024xf32>, %matC: memref<1024x1024xf32>) workgroup(%sharedA : memref<32x32xf32, #gpu.address_space<workgroup>>, %sharedB : memref<32x32xf32, #gpu.address_space<workgroup>>) kernel {
```

Comment:
I think there needs to be some comments in this example, pure code doesn't feel very good. Like how you did matmul,some details of the code.

Reply:
I have added some comments. However, the matmul case is still not easy to understand. The top answer to this [Stackoverflow question](https://stackoverflow.com/questions/18815489/cuda-tiled-matrix-matrix-multiplication-with-shared-memory-and-matrix-size-whic) described what I do.